### DISCUSSION TITLE
Go to top or bottom of a long list of siblings

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#708] [Ideas]
After a brainstorming session, I can end up with a very long list of siblings. I would like to be able to go to the top of the list or the bottom of the list with a keyboard and/or mouse command.

Right now, I create a dummy node, and if I'm at the bottom of the list, I press Ctrl+down, which moves the dummy node to the top of the list.  While at the top of the list, I can select the dummy node and press Ctrl+up and it moves that node to the bottom of the list. When I'm done, I delete the dummy node.

So FP _knows_ where the top and bottom of the siblings list is. I just have to use a funky way to go to these points.

To be very clear, I don't want to get rid of the current functionality.  I find it useful, too. But a cleaner keyboard shortcut would be nice.

~ Bal

**Comment by macmarrum** [comment]
Please feel free to use the following scripts and assign each a hot key of your choosing:
- [selectTopSibling.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/selectTopSibling.groovy)
- [selectBottomSibling.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/selectBottomSibling.groovy)

You can remove `="/menu_bar/Mac1/Select"` from the first line of each script to have it appear in `Tools->Scripts`

**Reply by wordmuse** [reply]
Fantastic - works perfectly! Thank you very much indeed, macmarrum (Mac?)!

~ Bal

**Comment by euu2021** [comment]
Maybe, I misunderstood your question, but I can jump to the top and bottom siblings (of the group of siblings) by pressing PageUp/PageDown.

**Reply by wordmuse** [reply]
No - you didn't. And I didn't know about these commands. 

That said, the commands and the scripts do behave slightly differently. The commands allow you to leave the list of siblings. If you're at the top of the list and press Page Up, it takes you above the list. Likewise, if you're at the bottom of the list of siblings, and you press Page Down, you leave the list.  The scripts constrain movement to the list of siblings. I thus consider the script to be an improvement for most of what I need it for.

Still, it's good to know about the commands. They don't show up when I do a command search. Are they in the menu system?

~ Bal

**Reply by euu2021** [reply]
> Are they in the menu system?

No, they are not.

**Reply by wordmuse** [reply]
I thought I knew FP pretty well, but clearly, I have more to learn. How did you come to know about these commands?

**Reply by euu2021** [reply]
Out of FP, I use PageUP/Down a lot to scroll through pages, so I think that trying those keys on FP was natural no me. On the other hand, until recently, I didn't know that they can be used to select cousins, as discussed as Edo pointed [here](https://github.com/freeplane/freeplane/discussions/677#discussioncomment-3729583).

So, yes, an interesting question is: how to make this kind of commands more naturally discoverable. Some time ago, I was thinking about the idea of bringing random tips in the splash screen. Game developers usually do this by inserting tips in the loading screen of games. If the user will need to stare the splash screen anyway, at least it can be used to teach something 😄

**Reply by wordmuse** [reply]
PageUp/PageDown no longer work - at least on my computer. I really did like having these commands. Fortunately the script provided by macmarrum still works and I can assign the keys to these scripts.

One more thing though: the native command keys that used work, but no longer does, also allowed a Shift key to be pressed so as to select all the siblings from the current node to either the top or the bottom of the siblings. The script doesn't have that.

**Reply by dpolivaev** [reply]
Have you checked the latest preview 1.11.1_31? It should work now.

**Reply by wordmuse** [reply]
Wonderful! Thanks, Dimitry!

~ Bal
